test/gtest/rocsparse/spmv_test.cpp

```diff
@@ -3,7 +3,12 @@
 #include <spblas/spblas.hpp>
 
 #include <gtest/gtest.h>
+
+#ifdef SPBLAS_ENABLE_ONEMKL_SYCL
+#include "onemkl/device_vector.hpp"
+#else
```

Comment:
so this is generalizing the rocsparse device spmv_test to just  a device/spmv_test.cpp for rocsparse/mkl_sycl and in the future others ?    Should we change the name from `rocsparse/`  folder to `device/`  or `accelerator/`  ?

Reply:
Yes, I put it into #40 (change it to device). I tended to avoid the same changes in two prs because it might make the reviews hard to exchange the idea in the same pr. I can move that into here in case this pr moves more quickly.

Reply:
can I suggest we shift it to `thrust_device`,  then if we have other device examples that are specific for say,  sycl or rocm or cuda we could have `sycl_device`, `rocm_device` or `cuda_device` folders as well ...

Reply:
Do different folders here contain the tests with different vector allocation or specific functions?
If you mean the specific functions, I might put them into `device/<vendor>`
`device` contains the uniform test for all backend, and `device/<vendor>` contains the tests just for specific vendor.
Both ways are clear, so I do not mind choosing another one unless one of them makes the CMake setting worse. (I do not think so, but just in case)